From 002178f7e60c1640749139f2561dff87b08c1b07 Mon Sep 17 00:00:00 2001
Message-Id: <002178f7e60c1640749139f2561dff87b08c1b07.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: =?UTF-8?q?Ville=20Syrj=C3=A4l=C3=A4?= <ville.syrjala@linux.intel.com>
Date: Sat, 28 Jun 2014 02:04:27 +0300
Subject: [PATCH 227/312] FROM_UPSTREAM [VPG]: drm/i915: Allow vdd_off when
 vdd is already off
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Allow calling the vdd off functions when vdd is already off. Makes
things simpler later.

Change-Id: I7e473a36224503c4c98d2ce3bea4717bd6156502
Signed-off-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Signed-off-by: Deepak S <deepak.s@intel.com>
Upstream-repo: http://patchwork.freedesktop.org/patch/28727/
Upstream-commit: <not applicable>
Upstream-as-of-tag: <not applicable>
---
 drivers/gpu/drm/i915/intel_dp.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index aefce46..e08b15c 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -1239,7 +1239,11 @@ static void edp_panel_vdd_off(struct intel_dp *intel_dp, bool sync)
 	if (!is_edp(intel_dp))
 		return;
 
-	WARN(!intel_dp->want_panel_vdd, "eDP VDD not forced on");
+	if (!edp_have_panel_vdd(intel_dp))
+		return;
+
+	if (!intel_dp->want_panel_vdd)
+		return;
 
 	intel_dp->want_panel_vdd = false;
 
-- 
1.7.9.5

